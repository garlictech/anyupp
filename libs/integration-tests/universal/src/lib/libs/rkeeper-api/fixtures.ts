import * as CrudApi from '@bgap/crud-gql/api';
import {
  Dish,
  RKeeperBusinessEntityInfo,
  RkeeperFixtures,
} from '@bgap/rkeeper-api';
import { RequiredId } from '@bgap/shared/types';
import {
  unitFixture,
  productFixture,
  chainFixture,
  groupFixture,
  rkeeperEndpoint,
  yellowRestaurantId,
  yellowRkeeperUsername,
  yellowRkeeperPassword,
} from '@bgap/shared/fixtures';

const testIdPrefix = `rkeeper-cf0d1110-a2ce-45cf-aa69-6782bbc44cad`;

export const rawData = {
  data: {
    dishes: [
      RkeeperFixtures.dish,
      RkeeperFixtures.dish2,
      RkeeperFixtures.duplicatedDish,
      RkeeperFixtures.badDish,
      RkeeperFixtures.inactiveDish,
      RkeeperFixtures.dishWithModifier,
      RkeeperFixtures.dishWithSameModifier,
    ],
    modifiers: [RkeeperFixtures.modifier, RkeeperFixtures.unusedModifier],
  },
};

export const rkeeperProductGuid = 'RKEEPERGUID';

const unitId = `${testIdPrefix}-unit`;
const chainId = `${testIdPrefix}-chain`;
const groupId = `${testIdPrefix}-group`;

export const rkeeperUnit: RequiredId<CrudApi.CreateUnitInput> = {
  ...unitFixture.createRkeeperUnit,
  id: unitId,
  groupId,
  chainId,
  externalId: 'EXTERNAL-RESTAURANT-ID',
};

export const rkeeperUnitProduct: RequiredId<CrudApi.CreateUnitProductInput> = {
  ...productFixture.unitProductInputBase,
  id: `${testIdPrefix}-unitproduct`,
  externalId: rkeeperProductGuid,
};

export const rkeeperUnitProduct2: RequiredId<CrudApi.CreateUnitProductInput> = {
  ...productFixture.unitProductInputBase,
  id: `${testIdPrefix}-unitproduct2`,
  externalId: rkeeperProductGuid + '2',
};

export const processedDish: Dish = {
  price: 50000,
  active: true,
  id: 9991000114,
  guid: '999e3ab3-86a0-48d9-a9a9-f4e0c9fbce68',
  name: 'Pr√≥ba ital',
};

export const businessEntity: RKeeperBusinessEntityInfo = {
  chainId,
  groupId,
  unitId,
};

export const createChain: RequiredId<CrudApi.CreateChainInput> = {
  ...chainFixture.chainBase,
  id: chainId,
};

export const createGroup: RequiredId<CrudApi.CreateGroupInput> = {
  ...groupFixture.groupBase,
  id: groupId,
  chainId,
};

export const rkeeperOrder = {
  objectid: yellowRestaurantId,
  remoteId: yellowRestaurantId,
  order_type: 1,
  pay_type: 0,
  pay_online_type: 0,
  delivery_time: '2122-02-03T12:37:32',
  client: {
    phone: null,
    email: null,
    ln: 'Testln',
    fn: 'Testfn',
  },
  order: [
    {
      id: '1000114',
      type: 'd',
      qnt: 1000,
    },
  ],
};

export const yellowUnit = {
  ...rkeeperUnit,
  externalId: yellowRestaurantId,
  pos: {
    ...rkeeperUnit.pos,
    rkeeper: {
      ...(rkeeperUnit?.pos?.rkeeper || {}),
      rkeeperUsername: yellowRkeeperUsername,
      rkeeperPassword: yellowRkeeperPassword,
      endpointUri: rkeeperEndpoint,
    } as CrudApi.RKeeper,
  },
} as CrudApi.Unit;

export const orderInput: CrudApi.CreateOrderInput = {
  userId: 'USER_ID',
  unitId: yellowUnit.id,
  items: [
    {
      quantity: 5,
      productId: 'PRODUCT ID',
      statusLog: [
        {
          userId: 'USER_ID',
          status: CrudApi.OrderStatus.none,
          ts: 1627909024677,
        },
      ],
      sumPriceShown: {
        taxSum: 316.98,
        currency: 'HUF',
        tax: 27,
        priceSum: 1491,
        pricePerUnit: 298.2,
      },
      productName: {
        hu: 'PRODUCT NAME HU',
      },
      priceShown: {
        taxSum: 318.9,
        currency: 'HUF',
        tax: 27,
        priceSum: 1500,
        pricePerUnit: 300,
      },
      variantId: 'VARIANT ID',
      variantName: {
        en: 'VARIANT NAME EN',
      },
    },
  ],
  sumPriceShown: {
    taxSum: 633.96,
    currency: 'HUF',
    tax: 27,
    priceSum: 2982,
    pricePerUnit: 298.2,
  },
  place: {
    table: '01',
    seat: '01',
  },
  orderMode: CrudApi.OrderMode.instant,
  servingMode: CrudApi.ServingMode.inplace,
  archived: false,
  paymentMode: {
    method: CrudApi.PaymentMethod.cash,
    type: CrudApi.PaymentType.cash,
  },
};

export {
  yellowRestaurantId,
  yellowRkeeperPassword,
  yellowRkeeperUsername,
  rkeeperEndpoint,
};
