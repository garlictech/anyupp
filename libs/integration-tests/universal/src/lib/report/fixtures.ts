import { DateTime } from 'luxon';

import { timezoneBudapest } from '@bgap/shared/utils';

const dateAgo = (days: number) =>
  DateTime.fromISO(new Date().toISOString(), {
    zone: timezoneBudapest,
  })
    .minus({ days })
    .toJSDate()
    .toString();

export const reportUsersFixture = [
  {
    id: 'anonym_1',
    email: 'testuser+anonymuser1@anyupp.com',
    ucd: new Date().getTime(),
  },
  {
    id: 'anonym_2',
    email: 'testuser+anonymuser2@anyupp.com',
    ucd: new Date().getTime(),
  },
  {
    id: 'anonym_3',
    email: 'testuser+anonymuser3@anyupp.com',
    ucd: new Date().getTime(),
  },
  {
    id: 'email_1',
    email: 'testuser+a1@anyupp.com',
    ucd: new Date().getTime(),
  },
  {
    id: 'email_2',
    email: 'testuser+a2@anyupp.com',
    ucd: new Date().getTime(),
  },
  {
    id: 'email_3',
    email: 'testuser+a3@anyupp.com',
    ucd: new Date().getTime(),
  },
  {
    id: 'facebook_1',
    email: 'testuser+fb1@anyupp.com',
    ucd: new Date().getTime(),
  },
  {
    id: 'facebook_2',
    email: 'testuser+fb2@anyupp.com',
    ucd: new Date().getTime(),
  },
  {
    id: 'google_1',
    email: 'testuser+g1@anyupp.com',
    ucd: new Date().getTime(),
  },
  {
    id: 'google_2',
    email: 'testuser+g2@anyupp.com',
    ucd: new Date().getTime(),
  },
  {
    id: 'signinwithapple_1',
    email: 'testuser+apple1@anyupp.com',
    ucd: new Date().getTime(),
  },
  {
    id: 'signinwithapple_2',
    email: 'testuser+apple2@anyupp.com',
    ucd: new Date().getTime(),
  },
];

export const fourWeeksOrdersFixture = [
  {
    createdAt: dateAgo(1),
    userId: 'anonym_1',
    sumPriceShown: { currency: 'HUF', priceSum: 1000 },
  },
  {
    createdAt: dateAgo(2),
    userId: 'anonym_1',
    sumPriceShown: { currency: 'HUF', priceSum: 1000 },
  },
  {
    createdAt: dateAgo(3),
    userId: 'anonym_2',
    sumPriceShown: { currency: 'HUF', priceSum: 1000 },
  },
  {
    createdAt: dateAgo(4),
    userId: 'anonym_3',
    sumPriceShown: { currency: 'HUF', priceSum: 1000 },
  },
  {
    createdAt: dateAgo(5),
    userId: 'email_1',
    sumPriceShown: { currency: 'HUF', priceSum: 1000 },
  },
  {
    createdAt: dateAgo(6),
    userId: 'email_2',
    sumPriceShown: { currency: 'HUF', priceSum: 1000 },
  },
  {
    createdAt: dateAgo(7),
    userId: 'email_3',
    sumPriceShown: { currency: 'HUF', priceSum: 1000 },
  },
  {
    createdAt: dateAgo(8),
    userId: 'email_3',
    sumPriceShown: { currency: 'HUF', priceSum: 1000 },
  },
  {
    createdAt: dateAgo(9),
    userId: 'facebook_1',
    sumPriceShown: { currency: 'HUF', priceSum: 1000 },
  },
  {
    createdAt: dateAgo(10),
    userId: 'facebook_2',
    sumPriceShown: { currency: 'HUF', priceSum: 1000 },
  },
];

export const thisYearOrdersFixture = [
  ...fourWeeksOrdersFixture,
  {
    createdAt: dateAgo(11),
    userId: 'facebook_2',
    sumPriceShown: { currency: 'HUF', priceSum: 1000 },
  },
  {
    createdAt: dateAgo(42),
    userId: 'google_1',
    sumPriceShown: { currency: 'HUF', priceSum: 1000 },
  },
  {
    createdAt: dateAgo(43),
    userId: 'google_2',
    sumPriceShown: { currency: 'HUF', priceSum: 1000 },
  },
  {
    createdAt: dateAgo(44),
    userId: 'signinwithapple_1',
    sumPriceShown: { currency: 'HUF', priceSum: 1000 },
  },
  {
    createdAt: dateAgo(45),
    userId: 'signinwithapple_2',
    sumPriceShown: { currency: 'HUF', priceSum: 1000 },
  },
];
