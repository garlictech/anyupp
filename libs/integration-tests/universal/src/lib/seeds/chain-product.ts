import { CrudSdk } from '@bgap/crud-gql/api';
import { CreateChainProductInput } from '@bgap/domain';
import { resultTap } from './seed.util';

export const createTestChainProduct = (
  input: CreateChainProductInput,
  crudSdk: CrudSdk,
) =>
  crudSdk
    .CreateChainProduct({ input })
    .pipe(resultTap('CHAIN PRODUCT create', input.id!));

export const deleteTestChainProduct = (id: string, crudSdk: CrudSdk) =>
  crudSdk
    .DeleteChainProduct({ input: { id } })
    .pipe(resultTap('CHAIN PRODUCT delete', id));
