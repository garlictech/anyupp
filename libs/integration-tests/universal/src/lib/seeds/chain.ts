import * as CrudApi from '@bgap/crud-gql/api';
import { resultTap } from './seed.util';

export const createTestChain = (
  input: CrudApi.CreateChainInput,
  crudSdk: CrudApi.CrudSdk,
) => crudSdk.CreateChain({ input }).pipe(resultTap('CHAIN create', input.id!));

export const deleteTestChain = (id: string, crudSdk: CrudApi.CrudSdk) =>
  crudSdk.DeleteChain({ input: { id } }).pipe(resultTap('CHAIN delete', id));
