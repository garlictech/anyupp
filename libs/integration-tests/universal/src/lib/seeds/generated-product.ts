import { CrudApi, CrudApiMutationDocuments } from '@bgap/crud-gql/api';
import {
  crudBackendGraphQLClient,
  executeMutation,
} from '@bgap/shared/graphql/api-client';
import { of } from 'rxjs';
import { catchError } from 'rxjs/operators';
import { resultTap } from './seed.util';

export const createTestGeneratedProduct = (
  input: CrudApi.CreateGeneratedProductInput,
) =>
  executeMutation(crudBackendGraphQLClient)<
    CrudApi.CreateGeneratedProductMutation
  >(CrudApiMutationDocuments.createGeneratedProduct, {
    input,
  }).pipe(resultTap('GENERATED_PRODUCT create', input.id!));

export const deleteTestGeneratedProduct = (id: string) =>
  executeMutation(crudBackendGraphQLClient)<
    CrudApi.DeleteGeneratedProductMutation
  >(CrudApiMutationDocuments.deleteGeneratedProduct, {
    input: { id },
  }).pipe(
    resultTap('GENERATED_PRODUCT delete', id),
    catchError(() => of('NO_PROBLEM')),
  );
