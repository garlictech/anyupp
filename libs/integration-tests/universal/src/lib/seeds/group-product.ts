import { CrudSdk } from '@bgap/crud-gql/api';
import { CreateGroupProductInput } from '@bgap/domain';
import { resultTap } from './seed.util';

export const createTestGroupProduct = (
  input: CreateGroupProductInput,
  crudSdk: CrudSdk,
) =>
  crudSdk
    .CreateGroupProduct({ input })
    .pipe(resultTap('GROUP PRODUCT create', input.id!));

export const deleteTestGroupProduct = (id: string, crudSdk: CrudSdk) =>
  crudSdk
    .DeleteGroupProduct({ input: { id } })
    .pipe(resultTap('GROUP PRODUCT delete', id));
