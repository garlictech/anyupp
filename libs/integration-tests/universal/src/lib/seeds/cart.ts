/* eslint-disable @typescript-eslint/no-non-null-assertion */
import { cartSeed } from '../fixtures/cart';
import { tap, catchError } from 'rxjs/operators';
import { throwError } from 'rxjs';
import {
  crudBackendGraphQLClient,
  executeMutation,
} from '@bgap/shared/graphql/api-client';
import { CrudApi, CrudApiMutationDocuments } from '@bgap/crud-gql/api';
import { resultTap } from './seed.util';

export const createTestCart = (input: Partial<CrudApi.CreateCartInput>) =>
  executeMutation(crudBackendGraphQLClient)<CrudApi.CreateCartMutation>(
    CrudApiMutationDocuments.createCart,
    { input },
  ).pipe(resultTap('CART create', input.id!));

export const deleteTestCart = (id: string) =>
  executeMutation(crudBackendGraphQLClient)<CrudApi.DeleteCartMutation>(
    CrudApiMutationDocuments.deleteCart,
    {
      input: { id },
    },
  ).pipe(resultTap('CART delete', id));
