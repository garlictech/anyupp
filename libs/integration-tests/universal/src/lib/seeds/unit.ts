/* eslint-disable @typescript-eslint/no-non-null-assertion */
import { unitSeed } from '../fixtures/unit';
import { tap, catchError } from 'rxjs/operators';
import { throwError } from 'rxjs';
import {
  crudBackendGraphQLClient,
  executeMutation,
} from '@bgap/shared/graphql/api-client';
import { CrudApi, CrudApiMutationDocuments } from '@bgap/crud-gql/api';
import { resultTap } from './seed.util';

export const createTestUnit = (input: Partial<CrudApi.CreateUnitInput>) =>
  executeMutation(crudBackendGraphQLClient)<CrudApi.CreateUnitMutation>(
    CrudApiMutationDocuments.createUnit,
    {
      input,
    },
  ).pipe(resultTap('UNIT create', input.id!));

export const deleteTestUnit = (id: string) =>
  executeMutation(crudBackendGraphQLClient)<CrudApi.DeleteUnitMutation>(
    CrudApiMutationDocuments.deleteUnit,
    { input: { id } },
  ).pipe(resultTap('UNIT delete', id));
