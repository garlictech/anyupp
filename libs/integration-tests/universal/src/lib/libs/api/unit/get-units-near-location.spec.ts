/* eslint-disable @typescript-eslint/no-explicit-any */
import { combineLatest, from } from 'rxjs';

import { AppsyncApi } from '@bgap/api/graphql/schema';
import { unitRequestHandler } from '@bgap/api/unit';
import {
  amplifyGraphQlClient,
  appsyncGraphQlClient,
  executeQuery,
} from '@bgap/shared/graphql/api-client';
import { unitSeed } from '../../../fixtures/unit';
import { createTestUnit, deleteTestUnit } from '../../../seeds/unit';
import { filter, map, switchMap } from 'rxjs/operators';
import * as fp from 'lodash/fp';

const userLoc = { location: { lat: 47.48992, lng: 19.046135 } }; // distance from seededUnitLoc: 54.649.. km
const distanceLoc_01 = { location: { lat: 47.490108, lng: 19.047077 } }; // distance from userLoc: 0.073.. km
const distanceLoc_02 = { location: { lat: 47.490471, lng: 19.048001 } }; // distance from userLoc: 0.153.. km
// const distanceLoc_03 = { location: { lat: 47.490877, lng: 19.04916 } }; // distance from userLoc: 0.250.. km
// const distanceLoc_04 = { location: { lat: 47.49121, lng: 19.050105 } }; // distance from userLoc: 0.330.. km
// const distanceLoc_05 = { location: { lat: 47.491979, lng: 19.05219 } }; // distance from userLoc: 0.509.. km
// const distanceLoc_06 = { location: { lat: 47.493168, lng: 19.055454 } }; // distance from userLoc: 0.787.. km

const unitNotActive = {
  isActive: false,
  id: 'NOT_ACTIVE_UNIT',
};

const unit_01 = {
  id: 'unit_01',
  address: fp.mergeAll([unitSeed.unit_01.address, distanceLoc_01]),
};
const unit_02 = {
  id: 'unit_02',
  address: fp.mergeAll([unitSeed.unit_01.address, distanceLoc_02]),
};
const unit_03 = {
  id: 'unit_03',
  address: fp.mergeAll([unitSeed.unit_01.address, userLoc]),
};

describe('GetUnitsNearLocation tests', () => {
  beforeAll(async () => {
    await combineLatest([
      // CleanUP
      deleteTestUnit(unitNotActive.id),
      deleteTestUnit(unit_01.id),
      deleteTestUnit(unit_02.id),
      deleteTestUnit(unit_03.id),
    ])
      .pipe(
        switchMap(() =>
          // Seeding
          combineLatest([
            createTestUnit(unitNotActive),
            createTestUnit(unit_01),
            createTestUnit(unit_02),
            createTestUnit(unit_03),
          ]),
        ),
      )
      .toPromise();
  });

  describe('input validation', () => {
    it('should throw without a input', done => {
      const input: AppsyncApi.GetUnitsNearLocationQueryVariables = {} as any;
      from(
        unitRequestHandler.getUnitsNearLocation(amplifyGraphQlClient)(input),
      ).subscribe({
        error(e) {
          expect(e).toMatchSnapshot();
          done();
        },
      });
    }, 15000);
    it('should throw without a location input', done => {
      const input: AppsyncApi.GetUnitsNearLocationQueryVariables = {
        input: {},
      } as any;
      from(
        unitRequestHandler.getUnitsNearLocation(amplifyGraphQlClient)(input),
      ).subscribe({
        error(e) {
          expect(e).toMatchSnapshot();
          done();
        },
      });
    }, 15000);
    it('should throw without a lat arg in the location input', done => {
      const input: AppsyncApi.GetUnitsNearLocationQueryVariables = {
        input: { location: { lat: '12' } },
      } as any;
      from(
        unitRequestHandler.getUnitsNearLocation(amplifyGraphQlClient)(input),
      ).subscribe({
        error(e) {
          expect(e).toMatchSnapshot();
          done();
        },
      });
    }, 15000);
    it('should throw without a lng arg in the location input', done => {
      const input: AppsyncApi.GetUnitsNearLocationQueryVariables = {
        input: { location: { lng: '12' } },
      } as any;
      from(
        unitRequestHandler.getUnitsNearLocation(amplifyGraphQlClient)(input),
      ).subscribe({
        error(e) {
          expect(e).toMatchSnapshot();
          done();
        },
      });
    }, 15000);
    it('should throw without valid location input', done => {
      const input: AppsyncApi.GetUnitsNearLocationQueryVariables = {
        input: { location: { lng: 230.0, lat: -100 } },
      } as any;

      executeQuery(appsyncGraphQlClient)<AppsyncApi.GetUnitsNearLocationQuery>(
        AppsyncApi.GetUnitsNearLocation,
        input,
      ).subscribe({
        error(e) {
          expect(e).toMatchSnapshot();
          done();
        },
      });
    }, 15000);
  });

  // TODO: create test with A NOT ACTIVE CHAIN
  it('should return all the units in geoUnitsFormat ordered by distance with direct handler call', done => {
    const input: AppsyncApi.GetUnitsNearLocationQueryVariables = {
      input: userLoc,
    };
    // To test with the local appsync code
    from(
      unitRequestHandler.getUnitsNearLocation(amplifyGraphQlClient)(input),
    ).subscribe({
      next(result) {
        expect(result).toHaveProperty('items');
        const foundItems: Array<AppsyncApi.GeoUnit> = result.items;
        successfullExecutionChecks(foundItems);
        done();
      },
    });
  }, 15000);

  // TODO: create test with A NOT ACTIVE CHAIN
  it('should return all the units in geoUnitsFormat ordered by distance with remove appsync api call', done => {
    const input: AppsyncApi.GetUnitsNearLocationQueryVariables = {
      input: userLoc,
    };
    executeQuery(appsyncGraphQlClient)<AppsyncApi.GetUnitsNearLocationQuery>(
      AppsyncApi.GetUnitsNearLocation,
      input,
    )
      .pipe(
        map(x => x.getUnitsNearLocation?.items),
        filter(x => !!x),
      )
      .subscribe({
        next(result) {
          if (!result || result === null) {
            throw 'Missing result';
          }
          // console.log(
          //   '### ~ file: get-units-near-location.spec.ts ~ line 88 ~ next ~ foundItems',
          //   JSON.stringify(result, undefined, 2),
          // );
          const foundItems: Array<AppsyncApi.GeoUnit> = result as Array<
            AppsyncApi.GeoUnit
          >;
          successfullExecutionChecks(foundItems);
          done();
        },
      });
  }, 15000);

  const successfullExecutionChecks = (
    foundItems: Array<AppsyncApi.GeoUnit>,
  ) => {
    const ids = foundItems.map(x => x.id);
    expect(ids).toContain(unitSeed.unitId_seeded_01);
    expect(ids).toContain(unitSeed.unitId_seeded_02);
    expect(ids).toContain(unitSeed.unitId_seeded_03);
    expect(ids).not.toContain(unitNotActive.id);

    expect(foundItems[0].id).toEqual(unit_03.id);
    expect(foundItems[1].id).toEqual(unit_01.id);
    expect(foundItems[2].id).toEqual(unit_02.id);
    expect(foundItems[0].distance).toEqual(0);
    expect(foundItems[1].distance).toEqual(74);
    expect(foundItems[2].distance).toEqual(153);
    // The rest is in the same location so we don't know their order
    expect(foundItems[3].distance).toEqual(54649);
    expect(foundItems[0]).toMatchSnapshot();
  };
});
