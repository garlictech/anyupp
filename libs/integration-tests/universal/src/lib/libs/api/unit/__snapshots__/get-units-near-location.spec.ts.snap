// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GetUnitsNearLocation tests input validation should throw without a input 1`] = `"GetUnitsNearLocationQueryVariables Object Validation Error: \\"input\\" is required"`;

exports[`GetUnitsNearLocation tests input validation should throw without a lat arg in the location input 1`] = `"GetUnitsNearLocationQueryVariables Object Validation Error: \\"input.location.lng\\" is required"`;

exports[`GetUnitsNearLocation tests input validation should throw without a lng arg in the location input 1`] = `"GetUnitsNearLocationQueryVariables Object Validation Error: \\"input.location.lat\\" is required"`;

exports[`GetUnitsNearLocation tests input validation should throw without a location input 1`] = `"GetUnitsNearLocationQueryVariables Object Validation Error: \\"input.location\\" is required"`;

exports[`GetUnitsNearLocation tests input validation should throw without valid location input 1`] = `[Error: GraphQL error: GetUnitsNearLocationQueryVariables Object Validation Error: "input.location.lat" must be greater than or equal to -90, "input.location.lng" must be less than or equal to 180]`;

exports[`GetUnitsNearLocation tests should return all the units in geoUnitsFormat ordered by distance with direct handler call 1`] = `
Object {
  "address": Object {
    "address": "Ág u. 1.",
    "city": "Budapest",
    "country": "Magyarország",
    "location": Object {
      "__typename": "Location",
      "lat": 47.48992,
      "lng": 19.046135,
    },
    "postalCode": "1021",
    "title": "HQ",
  },
  "chainId": "chain_1_id",
  "currency": "EUR",
  "distance": 0,
  "groupId": "group_c1_1_id",
  "id": "unit_03",
  "name": "Test unit #unit_1_id",
  "openingHours": "09:00 - 22:00",
  "paymentModes": Array [
    Object {
      "__typename": "PaymentMode",
      "caption": null,
      "method": "CASH",
      "name": "Cash",
    },
    Object {
      "__typename": "PaymentMode",
      "caption": null,
      "method": "CARD",
      "name": "Card",
    },
    Object {
      "__typename": "PaymentMode",
      "caption": null,
      "method": "INAPP",
      "name": "Stripe",
    },
  ],
  "style": Object {
    "colors": Object {
      "__typename": "ChainStyleColors",
      "backgroundDark": "#fff",
      "backgroundLight": "#fff",
      "borderDark": "#fff",
      "borderLight": "#fff",
      "disabled": "#fff",
      "highlight": "#fff",
      "indicator": "#fff",
      "textDark": "#fff",
      "textLight": "#fff",
    },
    "images": null,
  },
}
`;

exports[`GetUnitsNearLocation tests should return all the units in geoUnitsFormat ordered by distance with remove appsync api call 1`] = `
Object {
  "__typename": "GeoUnit",
  "address": Object {
    "__typename": "Address",
    "address": "Ág u. 1.",
    "city": "Budapest",
    "country": "Magyarország",
    "location": Object {
      "__typename": "Location",
      "lat": 47.48992,
      "lng": 19.046135,
    },
    "postalCode": "1021",
    "title": "HQ",
  },
  "chainId": "chain_1_id",
  "currency": "EUR",
  "distance": 0,
  "groupId": "group_c1_1_id",
  "id": "unit_03",
  "name": "Test unit #unit_1_id",
  "openingHours": "09:00 - 22:00",
  "paymentModes": Array [
    Object {
      "__typename": "PaymentMode",
      "caption": null,
      "method": "CASH",
      "name": "Cash",
    },
    Object {
      "__typename": "PaymentMode",
      "caption": null,
      "method": "CARD",
      "name": "Card",
    },
    Object {
      "__typename": "PaymentMode",
      "caption": null,
      "method": "INAPP",
      "name": "Stripe",
    },
  ],
  "style": Object {
    "__typename": "ChainStyle",
    "colors": Object {
      "__typename": "ChainStyleColors",
      "backgroundDark": "#fff",
      "backgroundLight": "#fff",
      "borderDark": "#fff",
      "borderLight": "#fff",
      "disabled": "#fff",
      "highlight": "#fff",
      "indicator": "#fff",
      "textDark": "#fff",
      "textLight": "#fff",
    },
    "images": null,
  },
}
`;
